### PR TITLE
Cover comands with UT

### PR DESCRIPTION
**Pull request by wolfylambova22**
Cover following commands with UTs:
- speak_request
- subscribe_button_request
- on_driver_distraction_notification
- get_vehicle_data_request
- delete_file_request
- on_device_state_changed_notification
- add_sub_menu_request
- subscribe_way_points_request

Related issue: APPLINK-28067

### CODE REVIEW

File: src/components/application_manager/test/commands/mobile/system_request_test.cc [hunk 1]
```diff
@@ -1,385 +0,0 @@
...
-      ManageMobileCommand(MobileResultCodeIs(mobile_result::SUCCESS), _));
-
-  command->Run();
-}
-
-}  // namespace system_request
-}  // namespace mobile_commands_test
-}  // namespace commands_test
-}  // namespace components
-}  // namespace test
```
Comment (LevchenkoS) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @wolfylambova22, why this file was deleted?

Comment (wolfylambova22) [inline src/components/application_manager/test/commands/mobile/system_request_test.cc]: @LevchenkoS it was pushed in this commit by mistake. UTs for this command will be included in PR for the related issue.

**Comment by wolfylambova22** [comment]
@AGritsevich PR is ready for merge.

File: src/components/application_manager/test/commands/hmi/sdl_activate_app_request_test.cc [hunk 2]
```diff
@@ -87,8 +87,7 @@ MATCHER_P2(CheckMsgParams, result, corr_id, "") {
 
   const bool is_corr_id_valid =
       corr_id ==
-      static_cast<int32_t>(
-          (*arg)[am::strings::params][am::strings::correlation_id].asInt());
+      (*arg)[am::strings::params][am::strings::correlation_id].asUInt();
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/sdl_activate_app_request_test.cc]: @wolfylambova22 Please use at least round braces here.

File: src/components/application_manager/test/commands/hmi/vr_is_ready_response_test.cc [hunk 3]
```diff
@@ -73,7 +73,8 @@ class VRIsReadyResponseTest : public CommandsTest<CommandsTestMocks::kIsNice> {
 };
 
 TEST_F(VRIsReadyResponseTest, RUN_SUCCESS) {
-  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  MessageSharedPtr command_msg(
+      CreateMessage(NsSmartDeviceLink::NsSmartObjects::SmartType_Map));
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/vr_is_ready_response_test.cc]: @wolfylambova22 Please create an alias for NsSmartDeviceLink::NsSmartObjects as smart_objects and use it. Like it was before.

File: src/components/application_manager/test/commands/hmi/vr_is_ready_response_test.cc [hunk 4]
```diff
@@ -73,7 +73,8 @@ class VRIsReadyResponseTest : public CommandsTest<CommandsTestMocks::kIsNice> {
...
   (*command_msg)[strings::msg_params][strings::number] = kStringNum;
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/hmi/vr_is_ready_response_test.cc]: @wolfylambova22 Also please add `using namespace strings` into your test cases. Therefore you can rewrite
`(*command_msg)[strings::params][strings::connection_key] = kConnectionKey;` to `(*command_msg)[params][connection_key] = kConnectionKey;`

File: src/components/application_manager/test/commands/mobile/subscribe_button_request_test.cc [hunk 5]
```diff
@@ -116,10 +122,13 @@ TEST_F(SubscribeButtonRequestTest, Run_SubscriptionNotAllowed_UNSUCCESS) {
...
 TEST_F(SubscribeButtonRequestTest, Run_UiIsNotSupported_UNSUCCESS) {
-  CommandPtr command(CreateCommand<SubscribeButtonRequest>());
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[am::strings::params][am::strings::connection_key] = kConnectionKey;
+  CommandPtr command(CreateCommand<SubscribeButtonRequest>(msg));
 
-  MockAppPtr app(CreateMockApp());
-  ON_CALL(mock_app_manager_, application(_)).WillByDefault(Return(app));
+  MockAppPtr mock_app(CreateMockApp());
+  ON_CALL(mock_app_manager_, application(kConnectionKey))
```
Comment (AByzhynar) [inline src/components/application_manager/test/commands/mobile/subscribe_button_request_test.cc]: @wolfylambova22 You use it almost in all test cases. May be it's worth to move it to SetUp or ctor? Or even create InitGetters() method and put all ON_CALLs there.

Comment (wolfylambova22) [inline src/components/application_manager/test/commands/mobile/subscribe_button_request_test.cc]: @AByzhynar  fixed in a751aee5dee55cc70778aa6c5ffc3a4c47da6a05.

File: src/components/application_manager/test/commands/hmi/vr_is_ready_response_test.cc (hunk 4 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/vr_is_ready_response_test.cc]: @AByzhynar  fixed in a751aee5dee55cc70778aa6c5ffc3a4c47da6a05.

File: src/components/application_manager/test/commands/hmi/vr_is_ready_response_test.cc (hunk 3 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/vr_is_ready_response_test.cc]: @AByzhynar  fixed in a751aee5dee55cc70778aa6c5ffc3a4c47da6a05.

File: src/components/application_manager/test/commands/hmi/sdl_activate_app_request_test.cc (hunk 2 above)
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/sdl_activate_app_request_test.cc]: @AByzhynar  fixed in a751aee5dee55cc70778aa6c5ffc3a4c47da6a05.

File: src/components/application_manager/test/commands/hmi/on_driver_distraction_notification_test.cc [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+
+TEST_F(HMIOnDriverDistractionNotificationTest,
+       Run_SendNotificationToMobile_SUCCESS) {
+  const uint32_t app_id = 1u;
+  const hmi_apis::Common_DriverDistractionState::eType state =
+      hmi_apis::Common_DriverDistractionState::DD_ON;
+  MessageSharedPtr msg = CreateMessage();
+  (*msg)[am::strings::msg_params][am::hmi_notification::state] = state;
+
+  NotificationPtr command(CreateCommand<OnDriverDistractionNotification>(msg));
```
Comment (wolfylambova22) [inline src/components/application_manager/test/commands/hmi/on_driver_distraction_notification_test.cc]: @AByzhynar  fixed in a751aee5dee55cc70778aa6c5ffc3a4c47da6a05.

**Comment by wolfylambova22** [comment]
@AGritsevich , please merge.

**Comment by AByzhynar** [comment]
@wolfylambova22 Please use The Past time in commit body (f.e. Deteled something ...) 
Add at least minimal description to the second commit what was done.

**Comment by wolfylambova22** [comment]
@AByzhynar according to article in Confluence https://adc.luxoft.com/confluence/display/APPLINK/Commit+in+git+-+the+best+practices we should use imperative mood in commit body.
Also my commit has description and it's short because only two lines are changed..

**Comment by AByzhynar** [comment]
@wolfylambova22 Read the article again. You have written not carefully.  And minimum related issue should be placed into commit body.

**Comment by wolfylambova22** [comment]
@AByzhynar ready.

**Pull request merged**